tasks/main.yml

```diff
@@ -1,3 +1,21 @@
+- name: "Install rsyslog-gnutls if using TLS (CentOS)"
+  yum:
+    name: "rsyslog-gnutls"
+    state: present
+  when: use_rsyslog_tls|bool == false and ansible_distribution == 'CentOS'
+  tags:
+    - rsyslog
+    - files
+
+- name: "Install rsyslog-gnutls if using TLS (Debian)"
+  apt:
+    name: "rsyslog-gnutls"
```

Comment:
There is some weird bugs with the versions in the ubuntu repos (8.16.0).
> unexpected GnuTLS error -50 in nsd_gtls.c:530: The request is invalid. [v8.16.0 try http://www.rsyslog.com/e/2078 ] 

You should install it from the PPA http://www.rsyslog.com/downloads/download-v8-stable/